src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,246 @@
+
+    """Loggly Collector."""
+
+    def get_default_config_help(self):
+        """Update configuration help."""
+        config_help = super(
+            LogglyCollector, self).get_default_config_help()
+        config_help.update({
+            'id': "Metric identifier",
+            'field': "Name of the field to retrieve, required",
+            'count': "Whether to retrieve the count of matching events",
+            'timestamp_field': "Name of the timestamp field."
+                               "Defaults to 'timestamp'",
+            'timestamp_format': "Timestamp format as for strptime or 'int'"
+                                "for unix time. Defaults to 'int'",
+            'subdomain': "Loggly subdomain to retrieve results from",
+            'query': "Loggly search query",
+            'username': "Loggly username",
+            'password': "Loggly password"
+        })
+        return config_help
+
+    def process_config(self):
+        """Prepare metrics' configuration."""
+        def set_defaults(config):
+            defaults = {
+                'timestamp_field': 'timestamp',
+                'timestamp_format': 'int',
+                'field': 'count',
+                'count': False,
+                'query': '*'
+            }
+            [config.update({key: value}) for key, value in defaults.items()
+                if key not in config.keys()]
+
+        super(LogglyCollector, self).process_config()
+
+        self.config_bunch = bunch.bunchify(self.config)
+        self.metrics_config = self.config_bunch.metric
+
+        if not all((item in self.config_bunch.keys()
+                    for item in ['subdomain', 'username', 'password'])):
+            raise DiamondException(
+                'All of subdomain, username and password must be specified '
+                'in file %s' % self.configfile)
+
+        for metric, config in self.metrics_config.items():
+            if 'field' not in config.keys():
+                raise DiamondException(
+                    'field must be specified for the metric %s in file %s' %
+                    (metric, self.configfile))
+
+            set_defaults(config)
+
+            config.update({
+                'search_URI': 'https://{1.subdomain}.loggly.com'
+                              '/apiv2/search?q={0.query}&from='
+                              '-{1.interval}s&until=now'.format(
+                                  config, self.config_bunch),
+                'events_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/events?rsid=%s'.format(
+                                  self.config_bunch)})
+
+    def collect(self):
+        """Collect metrics from loggly."""
+        def parse_timestamp(metric_config, timestamp):
+            if metric_config.timestamp_format is 'int':
+                return datetime.datetime.fromtimestamp(
+                    timestamp / 10**(ceil(log10(timestamp)) - 10))
```

Comment:
`strptime` doesn't provide format codes for epoch offsets, which is why we need to differentiate between `int` and any other `strptime`-like formats. The math for the `int` type is there to ensure we're passing seconds from epoch as we can't assume the offset timestamps are specified in seconds or any other time subdivision.